pkg/api/controller.go

```diff
@@ -225,7 +225,7 @@ func (c *Controller) InitImageStore(reloadCtx context.Context) error {
 	c.StoreController = storage.StoreController{}
 
 	if c.Config.Storage.RootDirectory != "" {
-		if c.Config.Storage.Dedupe {
+		if c.Config.Storage.Dedupe && len(c.Config.Storage.StorageDriver) == 0 {
```

Comment:
Why is it a `len()` check here?

Reply:
because we don't want the to check if s3 can do hard links, we have a check there that automatically disables dedupe if the underlying filesystem can't do hard links :D

Reply:
fixed